<Type Name="ForeignKeyReferenceAlreadyHasValueException" FullName="System.Data.Linq.ForeignKeyReferenceAlreadyHasValueException">
  <Metadata>
    <Meta Name="ms.openlocfilehash" Value="d8770cffaeaf250a5417fb744564f86f9759ee9f" />
    <Meta Name="ms.sourcegitcommit" Value="6a0b904069161bbaec4ffd02aa7d9cf38c61e72e" />
    <Meta Name="ms.translationtype" Value="HT" />
    <Meta Name="ms.contentlocale" Value="pt-BR" />
    <Meta Name="ms.lasthandoff" Value="06/24/2018" />
    <Meta Name="ms.locfileid" Value="36689295" />
  </Metadata>
  <TypeSignature Language="C#" Value="public class ForeignKeyReferenceAlreadyHasValueException : InvalidOperationException" />
  <TypeSignature Language="ILAsm" Value=".class public auto ansi beforefieldinit ForeignKeyReferenceAlreadyHasValueException extends System.InvalidOperationException" />
  <TypeSignature Language="DocId" Value="T:System.Data.Linq.ForeignKeyReferenceAlreadyHasValueException" />
  <TypeSignature Language="VB.NET" Value="Public Class ForeignKeyReferenceAlreadyHasValueException&#xA;Inherits InvalidOperationException" />
  <TypeSignature Language="C++ CLI" Value="public ref class ForeignKeyReferenceAlreadyHasValueException : InvalidOperationException" />
  <TypeSignature Language="F#" Value="type ForeignKeyReferenceAlreadyHasValueException = class&#xA;    inherit InvalidOperationException" />
  <AssemblyInfo>
    <AssemblyName>System.Data.Linq</AssemblyName>
    <AssemblyVersion>4.0.0.0</AssemblyVersion>
  </AssemblyInfo>
  <Base>
    <BaseTypeName>System.InvalidOperationException</BaseTypeName>
  </Base>
  <Interfaces />
  <Docs>
    <summary>
      <span data-ttu-id="7a5d2-101">Representa os erros que ocorrem quando é feita uma tentativa de alterar uma chave estrangeira quando a entidade já está carregada.</span>
      <span class="sxs-lookup">
        <span data-stu-id="7a5d2-101">Represents errors that occur when an attempt is made to change a foreign key when the entity is already loaded.</span>
      </span>
    </summary>
    <remarks>To be added.</remarks>
  </Docs>
  <Members>
    <MemberGroup MemberName=".ctor">
      <AssemblyInfo>
        <AssemblyName>System.Data.Linq</AssemblyName>
        <AssemblyVersion>4.0.0.0</AssemblyVersion>
      </AssemblyInfo>
      <Docs>
        <summary>
          <span data-ttu-id="7a5d2-102">Inicializa uma nova instância da classe <see cref="T:System.Data.Linq.ForeignKeyReferenceAlreadyHasValueException" />.</span>
          <span class="sxs-lookup">
            <span data-stu-id="7a5d2-102">Initializes a new instance of the <see cref="T:System.Data.Linq.ForeignKeyReferenceAlreadyHasValueException" /> class.</span>
          </span>
        </summary>
      </Docs>
    </MemberGroup>
    <Member MemberName=".ctor">
      <MemberSignature Language="C#" Value="public ForeignKeyReferenceAlreadyHasValueException ();" />
      <MemberSignature Language="ILAsm" Value=".method public hidebysig specialname rtspecialname instance void .ctor() cil managed" />
      <MemberSignature Language="DocId" Value="M:System.Data.Linq.ForeignKeyReferenceAlreadyHasValueException.#ctor" />
      <MemberSignature Language="VB.NET" Value="Public Sub New ()" />
      <MemberSignature Language="C++ CLI" Value="public:&#xA; ForeignKeyReferenceAlreadyHasValueException();" />
      <MemberType>Constructor</MemberType>
      <AssemblyInfo>
        <AssemblyName>System.Data.Linq</AssemblyName>
        <AssemblyVersion>4.0.0.0</AssemblyVersion>
      </AssemblyInfo>
      <Parameters />
      <Docs>
        <summary>
          <span data-ttu-id="7a5d2-103">Inicializa uma nova instância da classe <see cref="T:System.Data.Linq.ForeignKeyReferenceAlreadyHasValueException" /> com uma mensagem fornecida pelo sistema que descreve o erro.</span>
          <span class="sxs-lookup">
            <span data-stu-id="7a5d2-103">Initializes a new instance of the <see cref="T:System.Data.Linq.ForeignKeyReferenceAlreadyHasValueException" /> class with a system-supplied message that describes the error.</span>
          </span>
        </summary>
        <remarks>
          <format type="text/markdown"><![CDATA[  
  
## Remarks  
 <span data-ttu-id="7a5d2-104">Este construtor inicializa o <xref:System.Exception.Message%2A> propriedade da nova instância para uma mensagem fornecida pelo sistema que descreve o erro, como "DefaultMessageDisplayedByParameterlessCtorWriterMustSupply".</span><span class="sxs-lookup"><span data-stu-id="7a5d2-104">This constructor initializes the <xref:System.Exception.Message%2A> property of the new instance to a system-supplied message that describes the error, such as "DefaultMessageDisplayedByParameterlessCtorWriterMustSupply".</span></span> <span data-ttu-id="7a5d2-105">Esta mensagem leva em conta a cultura do sistema atual.</span><span class="sxs-lookup"><span data-stu-id="7a5d2-105">This message takes into account the current system culture.</span></span>  
  
 <span data-ttu-id="7a5d2-106">A tabela a seguir mostra os valores de propriedade iniciais para uma instância de <xref:System.InsufficientMemoryException>.</span><span class="sxs-lookup"><span data-stu-id="7a5d2-106">The following table shows the initial property values for an instance of <xref:System.InsufficientMemoryException>.</span></span>  
  
|<span data-ttu-id="7a5d2-107">Propriedade</span><span class="sxs-lookup"><span data-stu-id="7a5d2-107">Property</span></span>|<span data-ttu-id="7a5d2-108">Valor</span><span class="sxs-lookup"><span data-stu-id="7a5d2-108">Value</span></span>|  
|--------------|-----------|  
|<xref:System.Exception.InnerException%2A>|<span data-ttu-id="7a5d2-109">`null`.</span><span class="sxs-lookup"><span data-stu-id="7a5d2-109">`null`.</span></span>|  
|<xref:System.Exception.Message%2A>|<span data-ttu-id="7a5d2-110">A cadeia de caracteres da mensagem de erro localizada.</span><span class="sxs-lookup"><span data-stu-id="7a5d2-110">The localized error message string.</span></span>|  
  
 ]]></format>
        </remarks>
      </Docs>
    </Member>
    <Member MemberName=".ctor">
      <MemberSignature Language="C#" Value="public ForeignKeyReferenceAlreadyHasValueException (string message);" />
      <MemberSignature Language="ILAsm" Value=".method public hidebysig specialname rtspecialname instance void .ctor(string message) cil managed" />
      <MemberSignature Language="DocId" Value="M:System.Data.Linq.ForeignKeyReferenceAlreadyHasValueException.#ctor(System.String)" />
      <MemberSignature Language="VB.NET" Value="Public Sub New (message As String)" />
      <MemberSignature Language="C++ CLI" Value="public:&#xA; ForeignKeyReferenceAlreadyHasValueException(System::String ^ message);" />
      <MemberSignature Language="F#" Value="new System.Data.Linq.ForeignKeyReferenceAlreadyHasValueException : string -&gt; System.Data.Linq.ForeignKeyReferenceAlreadyHasValueException" Usage="new System.Data.Linq.ForeignKeyReferenceAlreadyHasValueException message" />
      <MemberType>Constructor</MemberType>
      <AssemblyInfo>
        <AssemblyName>System.Data.Linq</AssemblyName>
        <AssemblyVersion>4.0.0.0</AssemblyVersion>
      </AssemblyInfo>
      <Parameters>
        <Parameter Name="message" Type="System.String" />
      </Parameters>
      <Docs>
        <param name="message">
          <span data-ttu-id="7a5d2-111">A mensagem que descreve a exceção.</span>
          <span class="sxs-lookup">
            <span data-stu-id="7a5d2-111">The message that describes the exception.</span>
          </span>
          <span data-ttu-id="7a5d2-112">O chamador desse construtor é necessário para garantir que essa cadeia de caracteres tenha sido localizada para a cultura do sistema atual.</span>
          <span class="sxs-lookup">
            <span data-stu-id="7a5d2-112">The caller of this constructor is required to ensure that this string has been localized for the current system culture.</span>
          </span>
        </param>
        <summary>
          <span data-ttu-id="7a5d2-113">Inicializa uma nova instância da classe <see cref="T:System.Data.Linq.ForeignKeyReferenceAlreadyHasValueException" /> com uma mensagem especificada que descreve o erro.</span>
          <span class="sxs-lookup">
            <span data-stu-id="7a5d2-113">Initializes a new instance of the <see cref="T:System.Data.Linq.ForeignKeyReferenceAlreadyHasValueException" /> class with a specified message that describes the error.</span>
          </span>
        </summary>
        <remarks>
          <format type="text/markdown"><![CDATA[  
  
## Remarks  
 <span data-ttu-id="7a5d2-114">A tabela a seguir mostra os valores de propriedade iniciais para uma instância de <xref:System.Data.Linq.ForeignKeyReferenceAlreadyHasValueException>.</span><span class="sxs-lookup"><span data-stu-id="7a5d2-114">The following table shows the initial property values for an instance of <xref:System.Data.Linq.ForeignKeyReferenceAlreadyHasValueException>.</span></span>  
  
|<span data-ttu-id="7a5d2-115">Propriedade</span><span class="sxs-lookup"><span data-stu-id="7a5d2-115">Property</span></span>|<span data-ttu-id="7a5d2-116">Valor</span><span class="sxs-lookup"><span data-stu-id="7a5d2-116">Value</span></span>|  
|--------------|-----------|  
|<xref:System.Exception.InnerException%2A>|<span data-ttu-id="7a5d2-117">`null`.</span><span class="sxs-lookup"><span data-stu-id="7a5d2-117">`null`.</span></span>|  
|<xref:System.Exception.Message%2A>|<span data-ttu-id="7a5d2-118">A cadeia de caracteres da mensagem de erro especificada em `message`.</span><span class="sxs-lookup"><span data-stu-id="7a5d2-118">The error message string specified in `message`.</span></span>|  
  
 ]]></format>
        </remarks>
      </Docs>
    </Member>
    <Member MemberName=".ctor">
      <MemberSignature Language="C#" Value="public ForeignKeyReferenceAlreadyHasValueException (string message, Exception innerException);" />
      <MemberSignature Language="ILAsm" Value=".method public hidebysig specialname rtspecialname instance void .ctor(string message, class System.Exception innerException) cil managed" />
      <MemberSignature Language="DocId" Value="M:System.Data.Linq.ForeignKeyReferenceAlreadyHasValueException.#ctor(System.String,System.Exception)" />
      <MemberSignature Language="VB.NET" Value="Public Sub New (message As String, innerException As Exception)" />
      <MemberSignature Language="C++ CLI" Value="public:&#xA; ForeignKeyReferenceAlreadyHasValueException(System::String ^ message, Exception ^ innerException);" />
      <MemberSignature Language="F#" Value="new System.Data.Linq.ForeignKeyReferenceAlreadyHasValueException : string * Exception -&gt; System.Data.Linq.ForeignKeyReferenceAlreadyHasValueException" Usage="new System.Data.Linq.ForeignKeyReferenceAlreadyHasValueException (message, innerException)" />
      <MemberType>Constructor</MemberType>
      <AssemblyInfo>
        <AssemblyName>System.Data.Linq</AssemblyName>
        <AssemblyVersion>4.0.0.0</AssemblyVersion>
      </AssemblyInfo>
      <Parameters>
        <Parameter Name="message" Type="System.String" />
        <Parameter Name="innerException" Type="System.Exception" />
      </Parameters>
      <Docs>
        <param name="message">
          <span data-ttu-id="7a5d2-119">A mensagem que descreve a exceção.</span>
          <span class="sxs-lookup">
            <span data-stu-id="7a5d2-119">The message that describes the exception.</span>
          </span>
          <span data-ttu-id="7a5d2-120">O chamador desse construtor é necessário para garantir que essa cadeia de caracteres tenha sido localizada para a cultura do sistema atual.</span>
          <span class="sxs-lookup">
            <span data-stu-id="7a5d2-120">The caller of this constructor is required to ensure that this string has been localized for the current system culture.</span>
          </span>
        </param>
        <param name="innerException">
          <span data-ttu-id="7a5d2-121">A exceção que é a causa da exceção atual.</span>
          <span class="sxs-lookup">
            <span data-stu-id="7a5d2-121">The exception that is the cause of the current exception.</span>
          </span>
          <span data-ttu-id="7a5d2-122">Se o parâmetro <c>innerException</c> não for <see langword="null" />, a exceção atual será gerada em um bloco <see langword="catch" /> que tratará a exceção interna.</span>
          <span class="sxs-lookup">
            <span data-stu-id="7a5d2-122">If the <c>innerException</c> parameter is not <see langword="null" />, the current exception is raised in a <see langword="catch" /> block that handles the inner exception.</span>
          </span>
        </param>
        <summary>
          <span data-ttu-id="7a5d2-123">Inicializa uma nova instância da classe <see cref="T:System.Data.Linq.ForeignKeyReferenceAlreadyHasValueException" /> com uma mensagem de erro especificada e uma referência à exceção interna que é a causa da exceção.</span>
          <span class="sxs-lookup">
            <span data-stu-id="7a5d2-123">Initializes a new instance of the <see cref="T:System.Data.Linq.ForeignKeyReferenceAlreadyHasValueException" /> class with a specified error message and a reference to the inner exception that is the cause of this exception.</span>
          </span>
        </summary>
        <remarks>
          <format type="text/markdown"><![CDATA[  
  
## Remarks  
 <span data-ttu-id="7a5d2-124">Uma exceção que é lançada como um resultado direto de uma exceção anterior deve incluir uma referência para a exceção anterior na propriedade <xref:System.Exception.InnerException%2A>.</span><span class="sxs-lookup"><span data-stu-id="7a5d2-124">An exception that is thrown as a direct result of a previous exception should include a reference to the previous exception in the <xref:System.Exception.InnerException%2A> property.</span></span> <span data-ttu-id="7a5d2-125">A propriedade <xref:System.Exception.InnerException%2A> retorna o mesmo valor que é passado para o construtor, ou `null` caso a propriedade <xref:System.Exception.InnerException%2A> não forneça o valor de exceção interno para o construtor.</span><span class="sxs-lookup"><span data-stu-id="7a5d2-125">The <xref:System.Exception.InnerException%2A> property returns the same value that is passed into the constructor, or `null` if the <xref:System.Exception.InnerException%2A> property does not supply the inner exception value to the constructor.</span></span>  
  
 <span data-ttu-id="7a5d2-126">A tabela a seguir mostra os valores de propriedade iniciais para uma instância de <xref:System.Data.Linq.ForeignKeyReferenceAlreadyHasValueException>.</span><span class="sxs-lookup"><span data-stu-id="7a5d2-126">The following table shows the initial property values for an instance of <xref:System.Data.Linq.ForeignKeyReferenceAlreadyHasValueException>.</span></span>  
  
|<span data-ttu-id="7a5d2-127">Propriedade</span><span class="sxs-lookup"><span data-stu-id="7a5d2-127">Property</span></span>|<span data-ttu-id="7a5d2-128">Valor</span><span class="sxs-lookup"><span data-stu-id="7a5d2-128">Value</span></span>|  
|--------------|-----------|  
|<xref:System.Exception.InnerException%2A>|<span data-ttu-id="7a5d2-129">`null`.</span><span class="sxs-lookup"><span data-stu-id="7a5d2-129">`null`.</span></span>|  
|<xref:System.Exception.Message%2A>|<span data-ttu-id="7a5d2-130">A cadeia de caracteres da mensagem de erro especificada em `message`.</span><span class="sxs-lookup"><span data-stu-id="7a5d2-130">The error message string specified in `message`.</span></span>|  
  
 ]]></format>
        </remarks>
      </Docs>
    </Member>
  </Members>
</Type>